### PR TITLE
Implement serialization in rulesets

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

### Changed
- Rulesets can now be serialized and deserialized.
## Issue ticket number and link
NA

### CODE REVIEW

File: griptape/rules/base_rule.py [hunk 1]
```diff
@@ -5,10 +5,12 @@
 
 from attrs import define, field
 
+from griptape.mixins.serializable_mixin import SerializableMixin
 
-@define(frozen=True)
```
Comment (collindutter) [inline griptape/rules/base_rule.py:9]: Unfortunate but unavoidable without massive refactors.

Comment (dylanholmes) [reply griptape/rules/base_rule.py:9]: Would love a little more info on the thought process. If the goal is to make `value` and `meta` immutable is there any other ways? Like override the setter?

File: griptape/rules/rule.py [hunk 2]
```diff
@@ -5,9 +5,9 @@
 from griptape.rules import BaseRule
 
 
-@define(frozen=True)
+@define()
```
Comment (dylanholmes) [inline griptape/rules/rule.py:8]: Maybe remove parenthesis for consistency, obviously this is a really nit picky comment.

**Review by dylanholmes — APPROVED** [review]
No important comments.

**Comment by collindutter** [comment]
> Would love a little more info on the thought process. If the goal is to make value and meta immutable is there any other ways? Like override the setter?

@dylanholmes The change comes from pyright's note that:
```A frozen class cannot inherit from a class that is not frozen [reportGeneralTypeIssues]```

Making `SerializableMixin` frozen would be a massive (maybe unfeasable) change.

File: griptape/rules/base_rule.py (hunk 1 above)
Comment (collindutter) [reply griptape/rules/base_rule.py:9]: The change comes from pyright's note that:
```A frozen class cannot inherit from a class that is not frozen [reportGeneralTypeIssues]```

Making `SerializableMixin` frozen would be a massive (maybe unfeasable) change.

**Pull request merged**